azerothcore/azerothcore-wotlk
# refactor(Core): apply clang-tidy modernize-*

## Changes Proposed:
-  https://github.com/azerothcore/azerothcore-wotlk/discussions/9970

## Issues Addressed:

- None 


## Tests Performed:

- None



## How to Test the Changes:


These automated improvements are generally considered safe and there is no specific thing to check, but just in case:

1.  Compile and run the core
2. Access the game and try playing a little bit



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
